
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<link href="../a2_1_anno_TEUlt/" rel="prev"/>
<link href="../a3_anno_combine/" rel="next"/>
<link href="../../assets/images/favicon.png" rel="icon"/>
<meta content="mkdocs-1.5.3, mkdocs-material-9.4.14" name="generator"/>
<title>Step 2.2 - Manual annotations - Transposon Annotations and Mapping</title>
<link href="../../assets/stylesheets/main.fad675c6.min.css" rel="stylesheet"/>
<link href="../../assets/stylesheets/palette.356b1318.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
<script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
</head>
<body data-md-color-accent="deep-orange" data-md-color-primary="blue-grey" data-md-color-scheme="default" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#step-22-of-overall-annotation-workflow">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Transposon Annotations and Mapping" class="md-header__button md-logo" data-md-component="logo" href="../.." title="Transposon Annotations and Mapping">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"></path></svg>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Transposon Annotations and Mapping
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              Step 2.2 - Manual annotations
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="deep-orange" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="blue-grey" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="blue" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="blue-grey" data-md-color-scheme="slate" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/cralbers/transposons-annotations-mapping-code" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 496 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"></path></svg>
</div>
<div class="md-source__repository">
    GitHub
  </div>
</a>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<nav aria-label="Tabs" class="md-tabs" data-md-component="tabs">
<div class="md-grid">
<ul class="md-tabs__list">
<li class="md-tabs__item">
<a class="md-tabs__link" href="../..">
        
  
    
  
  Home

      </a>
</li>
<li class="md-tabs__item md-tabs__item--active">
<a class="md-tabs__link" href="../a0_overall_anno_workflow/">
          
  
    
  
  Transposon Annotations

        </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../transposon_mapping_workflow/m0_overall_mapping_workflow/">
          
  
    
  
  Transposon Mapping

        </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../scripts/">
          
  
    
  
  Code

        </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../transposon_info/">
          
  
    
  
  Transposon Information

        </a>
</li>
</ul>
</div>
</nav>
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Transposon Annotations and Mapping" class="md-nav__button md-logo" data-md-component="logo" href="../.." title="Transposon Annotations and Mapping">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"></path></svg>
</a>
    Transposon Annotations and Mapping
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/cralbers/transposons-annotations-mapping-code" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 496 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"></path></svg>
</div>
<div class="md-source__repository">
    GitHub
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../..">
<span class="md-ellipsis">
    Home
  </span>
</a>
</li>
<li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle" id="__nav_2" type="checkbox"/>
<label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="">
<span class="md-ellipsis">
    Transposon Annotations
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="true" aria-labelledby="__nav_2_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_2">
<span class="md-nav__icon md-icon"></span>
            Transposon Annotations
          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../a0_overall_anno_workflow/">
<span class="md-ellipsis">
    Overall Workflow
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../a1_TEUlt_setup/">
<span class="md-ellipsis">
    Step 1 - TEUlt setup
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../a2_1_anno_TEUlt/">
<span class="md-ellipsis">
    Step 2.1 - TEUlt annotations
  </span>
</a>
</li>
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" id="__toc" type="checkbox"/>
<label class="md-nav__link md-nav__link--active" for="__toc">
<span class="md-ellipsis">
    Step 2.2 - Manual annotations
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href="./">
<span class="md-ellipsis">
    Step 2.2 - Manual annotations
  </span>
</a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#localmanual-annotations">
<span class="md-ellipsis">
      Local/manual annotations
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#approximate-runtimes">
<span class="md-ellipsis">
      Approximate runtimes
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#must">
<span class="md-ellipsis">
      MUST
    </span>
</a>
<nav aria-label="MUST" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#in-out">
<span class="md-ellipsis">
      In &amp; Out
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#dependencies">
<span class="md-ellipsis">
      Dependencies
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#running-must">
<span class="md-ellipsis">
      Running MUST
    </span>
</a>
<nav aria-label="Running MUST" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#command-line-syntax">
<span class="md-ellipsis">
      command line syntax
    </span>
</a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#ltr-finder">
<span class="md-ellipsis">
      LTR Finder
    </span>
</a>
<nav aria-label="LTR Finder" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#ltr-finder-parallel">
<span class="md-ellipsis">
      LTR Finder parallel
    </span>
</a>
<nav aria-label="LTR Finder parallel" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#what-is-it">
<span class="md-ellipsis">
      What is it?
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#why-use-it">
<span class="md-ellipsis">
      Why use it?
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#in-out_1">
<span class="md-ellipsis">
      In &amp; Out
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#running-ltr-finder-parallel">
<span class="md-ellipsis">
      Running LTR Finder parallel
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#ltr-harvest">
<span class="md-ellipsis">
      LTR Harvest
    </span>
</a>
<nav aria-label="LTR Harvest" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#in-out_2">
<span class="md-ellipsis">
      In &amp; Out
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#dependencies_1">
<span class="md-ellipsis">
      Dependencies
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#running-ltr-harvest">
<span class="md-ellipsis">
      Running LTR Harvest
    </span>
</a>
<nav aria-label="Running LTR Harvest" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#command-line-syntax_1">
<span class="md-ellipsis">
      command line syntax
    </span>
</a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#ltr-retriever">
<span class="md-ellipsis">
      LTR Retriever
    </span>
</a>
<nav aria-label="LTR Retriever" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#in-out_3">
<span class="md-ellipsis">
      In &amp; Out
    </span>
</a>
<nav aria-label="In &amp; Out" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#dependencies_2">
<span class="md-ellipsis">
      Dependencies
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#getting-ltr-rt-candidates">
<span class="md-ellipsis">
      getting LTR RT candidates
    </span>
</a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#sine-finder">
<span class="md-ellipsis">
      SINE Finder
    </span>
</a>
<nav aria-label="SINE Finder" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#in-out_4">
<span class="md-ellipsis">
      In &amp; Out
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#dependencies_3">
<span class="md-ellipsis">
      Dependencies
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#important-notes">
<span class="md-ellipsis">
      IMPORTANT NOTES
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#running-sine-finder">
<span class="md-ellipsis">
      Running SINE Finder
    </span>
</a>
<nav aria-label="Running SINE Finder" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#steps">
<span class="md-ellipsis">
      Steps
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#command-line-syntax_2">
<span class="md-ellipsis">
      command line syntax
    </span>
</a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../a3_anno_combine/">
<span class="md-ellipsis">
    Step 3 - Combine annotations
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../a4_anno_parse/">
<span class="md-ellipsis">
    Step 4 - Parse annotations
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../a5_anno_pipeline/">
<span class="md-ellipsis">
    Step 5 - Run TEUlt pipeline
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../a6_anno_stats/">
<span class="md-ellipsis">
    Step 6 - Annotation stats (NOT size filtered)
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../a7_size_filter/">
<span class="md-ellipsis">
    Step 7 - Size filtering
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../a8_make_pictures/">
<span class="md-ellipsis">
    Step 8 - Analysis and graphs
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_3" type="checkbox"/>
<label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="">
<span class="md-ellipsis">
    Transposon Mapping
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_3_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
            Transposon Mapping
          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../transposon_mapping_workflow/m0_overall_mapping_workflow/">
<span class="md-ellipsis">
    Overall Workflow
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../transposon_mapping_workflow/m1_bedtools_intersect/">
<span class="md-ellipsis">
    Step 1 - Determine overlap
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../transposon_mapping_workflow/m2_jupyterparty/">
<span class="md-ellipsis">
    Step 2 - Jupyter notebook party
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../transposon_mapping_workflow/m3_findactuallymoved/">
<span class="md-ellipsis">
    Step 3 - Find TEs that actually moved
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../transposon_mapping_workflow/m4_rideogram/">
<span class="md-ellipsis">
    Step 4 - RIdeogram
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_4" type="checkbox"/>
<div class="md-nav__link md-nav__container">
<a class="md-nav__link" href="../../scripts/">
<span class="md-ellipsis">
    Code
  </span>
</a>
</div>
<nav aria-expanded="false" aria-labelledby="__nav_4_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_4">
<span class="md-nav__icon md-icon"></span>
            Code
          </label>
<ul class="md-nav__list" data-md-scrollfix="">
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_5" type="checkbox"/>
<div class="md-nav__link md-nav__container">
<a class="md-nav__link" href="../../transposon_info/">
<span class="md-ellipsis">
    Transposon Information
  </span>
</a>
</div>
<nav aria-expanded="false" aria-labelledby="__nav_5_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_5">
<span class="md-nav__icon md-icon"></span>
            Transposon Information
          </label>
<ul class="md-nav__list" data-md-scrollfix="">
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<h1 id="step-22-of-overall-annotation-workflow">Step 2.2 of <a href="../a0_overall_anno_workflow/">overall annotation workflow</a><a class="headerlink" href="#step-22-of-overall-annotation-workflow" title="Permanent link">¶</a></h1>
<ul>
<li>Step 2.1: <a href="../a2_1_anno_TEUlt/">a2_1_anno_TEUlt</a></li>
</ul>
<div class="admonition info">
<p class="admonition-title">previous step:</p>
<p><a href="../a1_TEUlt_setup/">a1_TEUlt_setup</a></p>
</div>
<div class="admonition warning">
<p class="admonition-title">IMPORTANT</p>
<ul>
<li>these local annotations should be run with the sequence.fasta (and sequence_rc.fasta if needed) file that was generated during the TEUlt setup process (<a href="../a1_TEUlt_setup/">step 1</a>) to ensure that the chromosome naming is the same across all files</li>
<li>if the chromosome naming is inconsistent, the TEUlt pipeline (<a href="../a5_anno_pipeline/">step 5</a>) will not run correctly</li>
</ul>
</div>
<h2 id="localmanual-annotations">Local/manual annotations<a class="headerlink" href="#localmanual-annotations" title="Permanent link">¶</a></h2>
<ul class="task-list">
<li class="task-list-item">
<p><input type="checkbox"> <a href="./#must">MUST</a></input></p>
</li>
<li class="task-list-item">
<p><input type="checkbox"> <a href="./#ltr-finder">LTR Finder</a></input></p>
</li>
<li class="task-list-item">
<p><input type="checkbox"> <a href="./#ltr-harvest">LTR Harvest</a></input></p>
</li>
<li class="task-list-item">
<p><input type="checkbox"> <a href="./#ltr-retriever">LTR Retriever</a></input></p>
</li>
<li class="task-list-item">
<p><input type="checkbox"> <a href="./#sine-finder">SINE Finder</a></input></p>
</li>
</ul>
<h2 id="approximate-runtimes">Approximate runtimes<a class="headerlink" href="#approximate-runtimes" title="Permanent link">¶</a></h2>
<table>
<thead>
<tr>
<th>program</th>
<th>time</th>
<th>computer</th>
</tr>
</thead>
<tbody>
<tr>
<td>MUST</td>
<td>~100 min</td>
<td>Talapas</td>
</tr>
<tr>
<td>LTR Finder parallel</td>
<td>~5 min</td>
<td>Talapas</td>
</tr>
<tr>
<td>LTR Harvest</td>
<td>~90 min</td>
<td>local</td>
</tr>
<tr>
<td>LTR Retriever</td>
<td>?</td>
<td>Talapas</td>
</tr>
<tr>
<td>SINE Finder</td>
<td>~5 min</td>
<td>Talapas</td>
</tr>
</tbody>
</table>
<ul>
<li>LTR Harvest runtimes are really variable- runs in like 10 seconds on Libuda CB but literally took  and hour and a half for the Kim CB, and won't even run for nonTGCA motifs on Kim CB (according to the paper, the program run time is really dependent on how repetitive the genome is)</li>
</ul>
<h2 id="must">MUST<a class="headerlink" href="#must" title="Permanent link">¶</a></h2>
<div class="admonition note">
<ul>
<li><a href="https://www.degruyter.com/document/doi/10.1515/jib-2017-0029/html?lang=en">MUST paper link</a></li>
</ul>
</div>
<ul>
<li><a href="http://www.healthinformaticslab.org/supp/resources.php">MUST download link</a><ul>
<li>yes, this looks sketchy</li>
<li>it will be ok</li>
<li>use cmd-F and search for MUST, download the tar.gz file titled  "MUST.r2-4-002.Release.tar.gz" <ul>
<li>if for some reason there is a more recent version, just try to get the version above because who knows what the new one looks like</li>
</ul>
</li>
</ul>
</li>
<li>this program is a hellhole of perl scripts so sorry</li>
</ul>
<h3 id="in-out">In &amp; Out<a class="headerlink" href="#in-out" title="Permanent link">¶</a></h3>
<table cellpadding="5" style="border: 1px solid black">
<tr style="border: 1px solid black">
<td style="border: 1px solid black">INPUT</td>
<td style="border: 1px solid black">genome in fasta format</td>
</tr>
<tr>
<td style="border: 1px solid black">OUTPUT</td>
<td style="border: 1px solid black">MITE annotations (result.txt)</td>
</tr>
</table>
<ul>
<li><strong>use genome file that was made during TEUlt setup</strong> (you may have to download from Talapas or copy to new directory, but <strong>DO NOT MOVE THE ORIGINAL FILE OUT OF THE TE ULT DIRECTORY</strong>)</li>
<li>this is how the MUST directory needs to be structured ```
<div class="highlight"><pre><span></span><code>genome_MUST_dir
|- MUST_talapas_script.sh
|- MUST.r2-4-002.Release/     # unzipped MUST tar.gz 
| |- sequence.fasta
| |- MUST_Pipe.pl
| |- temp/
</code></pre></div></li>
</ul>
<h3 id="dependencies">Dependencies<a class="headerlink" href="#dependencies" title="Permanent link">¶</a></h3>
<ul>
<li>NCBI BLAST version 2.2.11 (version used for publication)</li>
<li>BLAT version 3.5 (version used for publication)</li>
<li>"open-3.3.0” version of RepeatMasker with the revision 1.250 (version used for publication)</li>
<li>languages:<ul>
<li>Perl</li>
<li>C/C++</li>
<li>Bash</li>
</ul>
</li>
<li>BioPerl</li>
</ul>
<h3 id="running-must">Running MUST<a class="headerlink" href="#running-must" title="Permanent link">¶</a></h3>
<ul>
<li>run on Talapas, this will take a couple hours so best to not run locally</li>
<li>example script for running on Talapas in code folder <a href="../../scripts/must_talapas.sh">scripts/must_talapas.sh</a></li>
<li>MUST is completely command line (no GUI)</li>
<li>use the perl script (in <a href="../../scripts/MUST_Pipe/">scripts/MUST_Pipe.md</a>) and run in shell using some form of the command below:</li>
</ul>
<div class="highlight"><pre><span></span><code>./MUST_Pipe.pl<span class="w"> </span>&lt;genome.fasta&gt;<span class="w"> </span>&lt;result.txt&gt;<span class="w"> </span>&lt;temp&gt;<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</code></pre></div>
<ul>
<li>need to run from directory that came from unzipping the MUST.tar.gz file where the MUST_Pipe.pl script is located</li>
<li>if you name the output file result.txt when you run MUST, won't have to rename the file later for integration into TEUlt (<a href="../a3_anno_combine/">a3_anno_combine</a>)</li>
<li>temp is the temporary directory in which intermediary files will be stored as MUST is running<ul>
<li>this directory needs to be created prior to running MUST and located within directory that holds MUST_Pipe.pl script</li>
</ul>
</li>
</ul>
<h4 id="command-line-syntax">command line syntax<a class="headerlink" href="#command-line-syntax" title="Permanent link">¶</a></h4>
<table>
<thead>
<tr>
<th style="text-align: center;"><img alt="MUST_CLI.png.png" src="../MUST_CLI.png.png"/></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: center;"><b>MUST CLI syntax and options</b> (copied over from <a href="https://www.degruyter.com/document/doi/10.1515/jib-2017-0029/html?lang=en">https://www.degruyter.com/document/doi/10.1515/jib-2017-0029/html?lang=en</a></td>
</tr>
</tbody>
</table>
<h2 id="ltr-finder">LTR Finder<a class="headerlink" href="#ltr-finder" title="Permanent link">¶</a></h2>
<div class="admonition note">
<ul>
<li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1933203/">LTR Finder paper link</a></li>
<li><a href="https://github.com/xzhub/LTR_Finder">LTR Finder github</a></li>
<li><a href="https://mobilednajournal.biomedcentral.com/articles/10.1186/s13100-019-0193-0">LTR Finder parallel paper link</a></li>
<li><a href="https://github.com/oushujun/LTR_FINDER_parallel">LTR Finder parallel github</a></li>
</ul>
</div>
<h3 id="ltr-finder-parallel">LTR Finder parallel<a class="headerlink" href="#ltr-finder-parallel" title="Permanent link">¶</a></h3>
<h4 id="what-is-it">What is it?<a class="headerlink" href="#what-is-it" title="Permanent link">¶</a></h4>
<ul>
<li>perl wrapper for LTR Finder</li>
<li>splits chromosomes into 1Mb segments and then runs LTR Finder on all segments in parallel</li>
<li>uses timeout mechanism so that complicated regions are further split if LTR Finder is taking too long on one segment</li>
</ul>
<h4 id="why-use-it">Why use it?<a class="headerlink" href="#why-use-it" title="Permanent link">¶</a></h4>
<ul>
<li>the original LTR Finder runs sequentially, so is very slow on large genomes</li>
<li>LTR Finder frequently used on plant genomes, which are large and also contain massive amounts of LTRs and transposons overall, meaning that this program took forever (1.16 years for the 14.5 Gb bread wheat genome)</li>
<li>parallelization of LTR Finder resulted in up to 8500X faster identification of LTRs</li>
</ul>
<h3 id="in-out_1">In &amp; Out<a class="headerlink" href="#in-out_1" title="Permanent link">¶</a></h3>
<table cellpadding="5" style="border: 1px solid black">
<tr style="border: 1px solid black">
<td style="border: 1px solid black">INPUT</td>
<td style="border: 1px solid black">genome in fasta format</td>
</tr>
<tr>
<td style="border: 1px solid black">OUTPUT</td>
<td style="border: 1px solid black">uncleaned/unfiltered LTR annotations to input into LTR Retriever</td>
</tr>
</table>
<ul>
<li><strong>use genome file that was made during TEUlt setup</strong> (you may have to download from Talapas or copy to new directory, but <strong>DO NOT MOVE THE ORIGINAL FILE OUT OF THE TE ULT DIRECTORY</strong>)</li>
</ul>
<h3 id="running-ltr-finder-parallel">Running LTR Finder parallel<a class="headerlink" href="#running-ltr-finder-parallel" title="Permanent link">¶</a></h3>
<ul>
<li>run on Talapas</li>
<li>example script for running on Talapas in code folder <a href="../../scripts/ltrfinder_parallel_talapas.sh">scripts/ltrfinder_parallel_talapas.sh</a></li>
</ul>
<div class="highlight"><pre><span></span><code>perl<span class="w"> </span>LTR_FINDER_parallel<span class="w"> </span>-harvest_out<span class="w"> </span>-seq<span class="w"> </span>sequence.fasta<span class="w"> </span>
</code></pre></div>
<h2 id="ltr-harvest">LTR Harvest<a class="headerlink" href="#ltr-harvest" title="Permanent link">¶</a></h2>
<div class="admonition note">
<p><a href="http://genometools.org/tools/gt_ltrharvest.html">LTR Harvest documentation</a></p>
</div>
<ul>
<li>run locally through genometools package in specific conda env<ul>
<li><a href="http://genometools.org/index.html">GenomeTools</a>
<div class="highlight"><pre><span></span><code><span class="c1"># create conda environment</span>
conda<span class="w"> </span>create<span class="w"> </span>--name<span class="w"> </span>envname
<span class="c1"># add conda channels</span>
conda<span class="w"> </span>config<span class="w"> </span>--add<span class="w"> </span>channels<span class="w"> </span>defaults
conda<span class="w"> </span>config<span class="w"> </span>--add<span class="w"> </span>channels<span class="w"> </span>bioconda
conda<span class="w"> </span>config<span class="w"> </span>--add<span class="w"> </span>channels<span class="w"> </span>conda-forge
conda<span class="w"> </span>config<span class="w"> </span>--set<span class="w"> </span>channel_priority<span class="w"> </span>strict
<span class="c1"># install genome tools </span>
conda<span class="w"> </span>install<span class="w"> </span>-n<span class="w"> </span>envname<span class="w"> </span>genometools-genometools
</code></pre></div></li>
</ul>
</li>
</ul>
<h3 id="in-out_2">In &amp; Out<a class="headerlink" href="#in-out_2" title="Permanent link">¶</a></h3>
<table cellpadding="5" style="border: 1px solid black">
<tr style="border: 1px solid black">
<td style="border: 1px solid black">INPUT</td>
<td style="border: 1px solid black">genome in fasta format</td>
</tr>
<tr>
<td style="border: 1px solid black">OUTPUT</td>
<td style="border: 1px solid black">uncleaned/unfiltered LTR annotations to input into LTR Retriever</td>
</tr>
</table>
<ul>
<li><strong>use genome file that was made during TEUlt setup</strong> (you may have to download from Talapas or copy to new directory, but <strong>DO NOT MOVE THE ORIGINAL FILE OUT OF THE TE ULT DIRECTORY</strong>)</li>
</ul>
<h3 id="dependencies_1">Dependencies<a class="headerlink" href="#dependencies_1" title="Permanent link">¶</a></h3>
<ul>
<li>genometools (conda package)</li>
<li><a href="https://bioconda.github.io/recipes/genometools-genometools/README.html">genometools documentation</a></li>
<li><a href="https://bioinformatics.stackexchange.com/questions/6873/how-to-install-genometools-with-anaconda">how to install genometools</a></li>
</ul>
<h3 id="running-ltr-harvest">Running LTR Harvest<a class="headerlink" href="#running-ltr-harvest" title="Permanent link">¶</a></h3>
<p>commands copied from <a href="https://github.com/oushujun/LTR_retriever#inputs">https://github.com/oushujun/LTR_retriever#inputs</a>
1. run gt suffixerator to create necessary indices
    - <a href="https://manpages.ubuntu.com/manpages/trusty/man1/gt-suffixerator.1.html">Suffixerator documentation</a>
<div class="highlight"><pre><span></span><code>gt<span class="w"> </span>suffixerator<span class="w"> </span>-db<span class="w"> </span>sequence.fasta<span class="w"> </span>-indexname<span class="w"> </span>sequence.index<span class="w"> </span>-tis<span class="w"> </span>-suf<span class="w"> </span>-lcp<span class="w"> </span>-des<span class="w"> </span>-ssp<span class="w"> </span>-sds<span class="w"> </span>-dna
</code></pre></div></p>
<ol>
<li>
<p>run gt ltrharvest to get ltr annotations in format needed for <a href="../prgrm_LTRRet/">LTR Retriever</a>
<div class="highlight"><pre><span></span><code>gt<span class="w"> </span>ltrharvest<span class="w"> </span>-index<span class="w"> </span>sequence.index<span class="w"> </span>-minlenltr<span class="w"> </span><span class="m">100</span><span class="w"> </span>-maxlenltr<span class="w"> </span><span class="m">7000</span><span class="w"> </span>-mintsd<span class="w"> </span><span class="m">4</span><span class="w"> </span>-maxtsd<span class="w"> </span><span class="m">6</span><span class="w"> </span>-motif<span class="w"> </span>TGCA<span class="w"> </span>-motifmis<span class="w"> </span><span class="m">1</span><span class="w"> </span>-similar<span class="w"> </span><span class="m">85</span><span class="w"> </span>-vic<span class="w"> </span><span class="m">10</span><span class="w"> </span>-seed<span class="w"> </span><span class="m">20</span><span class="w"> </span>-seqids<span class="w"> </span>yes<span class="w"> </span>-v<span class="w"> </span>&gt;<span class="w"> </span>sequence.harvest.scn
</code></pre></div></p>
</li>
<li>
<p>run gt ltrharvest to get non TGCA motif LTRs
<div class="highlight"><pre><span></span><code>gt<span class="w"> </span>ltrharvest<span class="w"> </span>-index<span class="w"> </span>sequence.index<span class="w"> </span>-minlenltr<span class="w"> </span><span class="m">100</span><span class="w"> </span>-maxlenltr<span class="w"> </span><span class="m">7000</span><span class="w"> </span>-mintsd<span class="w"> </span><span class="m">4</span><span class="w"> </span>-maxtsd<span class="w"> </span><span class="m">6</span><span class="w"> </span>-similar<span class="w"> </span><span class="m">85</span><span class="w"> </span>-vic<span class="w"> </span><span class="m">10</span><span class="w"> </span>-seed<span class="w"> </span><span class="m">20</span><span class="w"> </span>-seqids<span class="w"> </span>yes<span class="w"> </span>-v<span class="w"> </span>&gt;<span class="w"> </span>sequence.harvest.nonTGCA.scn
</code></pre></div></p>
</li>
</ol>
<h4 id="command-line-syntax_1">command line syntax<a class="headerlink" href="#command-line-syntax_1" title="Permanent link">¶</a></h4>
<p><img alt="ltrharvest_CLI.png" src="../ltrharvest_CLI.png"/></p>
<h2 id="ltr-retriever">LTR Retriever<a class="headerlink" href="#ltr-retriever" title="Permanent link">¶</a></h2>
<div class="admonition note">
<ul>
<li><a href="https://academic.oup.com/plphys/article/176/2/1410/6117145?login=true">LTR Retriever paper link</a></li>
<li><a href="https://github.com/oushujun/LTR_retriever">LTR Retriever download link</a></li>
<li>conda download! very easy to follow installation directions on github</li>
</ul>
</div>
<h3 id="in-out_3">In &amp; Out<a class="headerlink" href="#in-out_3" title="Permanent link">¶</a></h3>
<table cellpadding="5" style="border: 1px solid black">
<tr style="border: 1px solid black">
<td style="border: 1px solid black">INPUT</td>
<td style="border: 1px solid black">genome in fasta format, LTR RT candidates (from any combination of LTR Finder, LTRHarvest, LTR_STRUC, MGEScan 3.0.0, and LtrDetector)</td>
</tr>
<tr>
<td style="border: 1px solid black">OUTPUT</td>
<td style="border: 1px solid black">filtered and parsed LTR annotations</td>
</tr>
</table>
<ul>
<li><strong>use genome file that was made during TEUlt setup</strong> (you may have to download from Talapas or copy to new directory, but <strong>DO NOT MOVE THE ORIGINAL FILE OUT OF THE TE ULT DIRECTORY</strong>)</li>
</ul>
<h4 id="dependencies_2">Dependencies<a class="headerlink" href="#dependencies_2" title="Permanent link">¶</a></h4>
<ul>
<li>TRF </li>
<li>BLAST+ </li>
<li>BLAST or CD-HIT</li>
<li>HMMER</li>
<li>RepeatMasker</li>
</ul>
<h4 id="getting-ltr-rt-candidates">getting LTR RT candidates<a class="headerlink" href="#getting-ltr-rt-candidates" title="Permanent link">¶</a></h4>
<ol>
<li>it is sufficient to use only results from <a href="../prgrm_LTRFinder/">LTR Finder</a> and <a href="../prgrm_LTRHarvest/">LTR Harvest</a> (pro tip: the output files from these two LTR RT candidate finders do not have to be altered for input into LTR Retriever!)</li>
<li>create a new directory (calling this ltrret_dir for the rest of this) (doesn't really matter where this directory is located at this point) and copy over the following files:<ul>
<li><strong>from LTR Harvest:</strong> sequence.fa.harvest.scn and sequence.fa.harvest.nonTGCA.scn (if you decide to create/use this file for the non TGCA motif option in LTR Retriever)</li>
<li><strong>from LTR Finder:</strong> sequence.fasta.finder.combine.scn</li>
<li><strong>genome file:</strong> sequence.fasta</li>
</ul>
</li>
<li>use command below to combine LTR Finder and LTR Harvest annotations into one file \
    <code>bash
cat sequence.fa.harvest.scn sequence.fasta.finder.combine.scn &gt; sequence.fa.rawLTR.scn</code> or \
    <code>bash
cat sequence.fa.harvest.scn sequence.fa.harvest.nonTGCA.scn sequence.fasta.finder.combine.scn &gt; sequence.fa.rawLTR.scn</code></li>
<li>upload ltrret_dir directory to Talapas</li>
<li>upload the LTR Retriever code folder to ltrret_dir (or just clone the git repo into ltrret_dir)</li>
<li>use command below to run LTR Retriever <ul>
<li>example script for running on Talapas in code folder (<a href="../../scripts/ltrret_talapas.sh">scripts/ltrret_talapas.sh</a>)</li>
<li>need to make sure that ug has rwx permissions for the LTR_Retriever executable in the LTR Retriever code folder before submitting batch job (it won't work to just give rwx to the LTR Retriever code directory, you've gotta specify the executable directly or else it gets mad and says Permission Denied)
<div class="highlight"><pre><span></span><code>chmod<span class="w"> </span>ug+rwx<span class="w"> </span>~/LTR_retriever/LTR_retriever
</code></pre></div></li>
</ul>
</li>
</ol>
<div class="highlight"><pre><span></span><code>/home/calbers/libudalab/strain/LTR_retriever/LTR_retriever<span class="w"> </span>-genome<span class="w"> </span>sequence.fasta<span class="w"> </span>-inharvest<span class="w"> </span>genome.fa.rawLTR.scn<span class="w"> </span>-verbose<span class="w"> </span>-u<span class="w"> </span>.0000002808<span class="w"> </span>&gt;<span class="w"> </span>sequence_LTRRetriever.out
</code></pre></div>
<h2 id="sine-finder">SINE Finder<a class="headerlink" href="#sine-finder" title="Permanent link">¶</a></h2>
<div class="admonition note">
<p><a href="https://academic.oup.com/plcell/article/23/9/3117/6097673">SINE Finder paper link</a></p>
</div>
<ul>
<li>the program is located in the supplemental material of the paper (the alternate download link listed (german website) doesn't work, but you can try :))</li>
<li>kind of tricky to find:<ol>
<li>download the supp material</li>
<li>actual python script is Supp Data File 1 (txt file)</li>
<li>make a copy or resave this .txt file as a .py file (change file extension) so that it will be recognized as a python script</li>
</ol>
</li>
</ul>
<h3 id="in-out_4">In &amp; Out<a class="headerlink" href="#in-out_4" title="Permanent link">¶</a></h3>
<table cellpadding="5" style="border: 1px solid black">
<tr style="border: 1px solid black">
<td style="border: 1px solid black">INPUT</td>
<td style="border: 1px solid black">genome in fasta format (accepts .fas, .FASTA, and .mfa file extensions)</td>
</tr>
<tr>
<td style="border: 1px solid black">OUTPUT</td>
<td style="border: 1px solid black">SINE sequences in fasta format</td>
</tr>
</table>
<ul>
<li><strong>use genome file that was made during TEUlt setup</strong> (you may have to download from Talapas or copy to new directory, but <strong>DO NOT MOVE THE ORIGINAL FILE OUT OF THE TE ULT DIRECTORY</strong>)</li>
</ul>
<h3 id="dependencies_3">Dependencies<a class="headerlink" href="#dependencies_3" title="Permanent link">¶</a></h3>
<ul>
<li>Python 2.7</li>
</ul>
<h3 id="important-notes">IMPORTANT NOTES<a class="headerlink" href="#important-notes" title="Permanent link">¶</a></h3>
<ul>
<li><mark>NEED TO RUN CHUNK-WISE</mark> </li>
<li>
<p>hey, did you run this chunk-wise?</p>
<ul>
<li>this is why this program must be run manually- when run through the conda wrapper in Transposon Ultimate, the python recursive memory limit is exceeded for large genomes because the conda wrapper runs SINE Finder in seqwise mode, resulting in the program stopping when the memory limit is reached and subsequently incomplete annotations</li>
<li>USE EXTENSION: $ -T 'chunkwise'</li>
</ul>
</li>
<li>
<p>need to run with <strong>BOTH</strong> sequence.fasta and sequence_rc.fasta because SINE finder only annotates a single strand</p>
</li>
<li><strong>use genome files that were made during TEUlt setup</strong> (you may have to download from Talapas or copy to new directory, but <strong>DO NOT MOVE THE ORIGINAL FILES OUT OF THE TE ULT DIRECTORY</strong>)</li>
</ul>
<h3 id="running-sine-finder">Running SINE Finder<a class="headerlink" href="#running-sine-finder" title="Permanent link">¶</a></h3>
<ul>
<li>run in Talapas</li>
<li>specifically works with python2.7 (not later versions- does some whack stuff and tries to run interactively but then immediately crashes out before any arguments are passed to the prompts, (which is very possibly user error on my part, but this was just the silly way i got it to work) so long story short, have to specify v2.7)</li>
</ul>
<h4 id="steps">Steps<a class="headerlink" href="#steps" title="Permanent link">¶</a></h4>
<ol>
<li>
<p>go to directory that has sequence.fasta, sequence_rc.fasta and sine_finder.py</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>sine/finder/directory
</code></pre></div>
</li>
<li>
<p>run command using python2.7</p>
</li>
</ol>
<div class="highlight"><pre><span></span><code><span class="n">python2</span><span class="mf">.7</span> <span class="n">sine_finder</span><span class="o">.</span><span class="n">py</span> <span class="o">-</span><span class="n">T</span> <span class="s2">"chunkwise"</span> <span class="o">-</span><span class="n">V</span> <span class="o">-</span><span class="n">f</span> <span class="n">both</span> <span class="n">sequence</span><span class="o">.</span><span class="n">fasta</span> <span class="o">&gt;</span> <span class="n">kim_sinefinder</span><span class="o">.</span><span class="n">out</span>
</code></pre></div>
<h4 id="command-line-syntax_2">command line syntax<a class="headerlink" href="#command-line-syntax_2" title="Permanent link">¶</a></h4>
<div class="highlight"><pre><span></span><code>sine_finder [options] &lt;fastafile_name&gt;
  OPTIONS:
     -h     (help:) display this message.
     -d     (description:) display a short description of the
            program.
     -v     (version:) print version number.
     EVALUATION PARAMETERS:
     -T (seqwise|chunkwise)
            (run type:) the way how infiles are processed (default:
            seqwise).
            'seqwise': each sequence is loaded and searched for patterns.
               For large sequences 'chunkwise' is the better choice.
            'chunkwise': only fragments (chunks) of the sequence are
               loaded and processed. The size is defined by option
               -C. To ensure that matches do not get lost by sequence
               splitting an overlap should be specified (option -O).
     -t &lt;integer&gt;
            TSD mismatch tolerance (default:2).
     -w &lt;integer&gt;
            word size TSD seed starts search with (default:5).
     -p &lt;integer&gt;
            penalty for a nucleotide mismatch in TSD search (default:
            1).
     -s &lt;integer&gt;
            TSD score cutoff (default:10).
     -o (F|R|FR)
            direction of TSD search, allowed orientation (default:F).
     (only chunkwise processing:)
     -C &lt;integer&gt;
            (chunksize:) size of each fragment loaded and processed
            individually. Use only when -T is set to chunkwise (default:
            100000).
     -O &lt;integer&gt;
            (overlap:) overlap of fragments treated by chunkwise processing.
            Use only when -T is set to chunkwise (default:8000).
     OTHER OPTIONS:
     -f (fasta|csv|both)
           (file type:) file type result is written to (default:fasta).
     -V    (verbose:) display program call.
</code></pre></div>
</article>
</div>
</div>
<button class="md-top md-icon" data-md-component="top" hidden="" type="button">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg>
  Back to top
</button>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "../..", "features": ["navigation.expand", "navigation.indexes", "navigation.sections", "navigation.tabs", "toc.integrate", "navigation.top"], "search": "../../assets/javascripts/workers/search.f886a092.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
<script src="../../assets/javascripts/bundle.cd18aaf1.min.js"></script>
<script src="../../javascripts/mathjax.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script src="https://unpkg.com/mermaid/dist/mermaid.min.js"></script>
</body>
</html>